robottelo/hosts.py

```diff
@@ -1299,3 +1299,22 @@ def update_setting(self, name, value):
         setting.value = value
         setting.update({'value'})
         return default_setting_value
+
+    def register_to_dogfood(self):
+        dogfood_canonical_hostname = settings.repos.dogfood_repo_host.partition('//')[2]
+        # get hostname of dogfood machine
+        dig_result = self.execute(f'dig +short {dogfood_canonical_hostname}')
+        # the host name finishes with a dot, so last character is removed
+        dogfood_hostname = dig_result.stdout.split()[0][:-1]
+        dogfood = Satellite(dogfood_hostname)
+        self.install_katello_ca(satellite=dogfood)
+        # satellite version consist from x.y.z, we need only x.y
+        sat_release = '.'.join(self.version.split('.')[:-1])
+        cmd_result = self.register_contenthost(
+            org=f'{settings.subscription.dogfood_org}',
+            activation_key=f'satellite-{sat_release}-qa-rhel7',
+        )
+        if cmd_result.status != 0:
+            raise SatelliteHostError(
+                f'Error during registration, command output: {cmd_result.stdout}'
+            )
```

Comment:
now we have this code here twice 

I believe the implementation I created in 
https://github.com/SatelliteQE/robottelo/blob/master/pytest_fixtures/core/sys.py#L47-L61
should be replaced by the method you introduced

It will look like this:
```
@pytest.fixture
def register_sat_to_dogfood(default_sat):
default_sat.register_to_dogfood()
```
rename the fixture name and use the method you implemented

Reply:
Yes, I saw your implementation earlier, I reused the very same code here for the helper, I missed updating this earlier, thanks for reminding me :)
Updated herein same PR in f35dcb7cf03627ce92da1c6063687704f9aeed17

Reply:
@tstrych  Also, tested your cockpit test locally, the satellite is registered to dogfood, and it enables an installer flag to enable cockpit successfully, but the test fails with an error below, which seems not related to this PR
```E               selenium.common.exceptions.NoSuchElementException: Message: Failed waiting for element with //div[@id="index-brand"] in None```

Reply:
thanks for running the_cockpit, I will look into that error, once this is merged :)